src/Directory.Build.props

```diff
@@ -1,29 +1,47 @@
 <Project>
-  <PropertyGroup>
+
+  <PropertyGroup Label="Build configuration">
+    <LangVersion>7.3</LangVersion>
     <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
     <BaseIntermediateOutputPath>$(MSBuildThisFileDirectory)..\obj\$(MSBuildProjectName)\</BaseIntermediateOutputPath>
     <OutputPath>$(MSBuildThisFileDirectory)..\bin\$(MSBuildProjectName)\$(Configuration)\</OutputPath>
     <PackageOutputPath>$(MSBuildThisFileDirectory)..\bin\Packages\$(Configuration)\</PackageOutputPath>
+    <IsTestProject>$(MSBuildProjectName.Contains('Test'))</IsTestProject>
+    <IsPackable Condition=" $(IsTestProject) ">false</IsPackable>
+    <GenerateDocumentationFile Condition=" !$(IsTestProject) ">true</GenerateDocumentationFile>
+    <CodeAnalysisRuleSet Condition=" !$(IsTestProject) ">$(MSBuildThisFileDirectory)/CodeGeneration.Roslyn.ruleset</CodeAnalysisRuleSet>
+    <CodeAnalysisRuleSet Condition=" $(IsTestProject) ">$(MSBuildThisFileDirectory)/CodeGeneration.Roslyn.Tests.ruleset</CodeAnalysisRuleSet>
+    <WarningsAsErrors>true</WarningsAsErrors>
```

Comment:
Not here, please. Warnings should *not* break the build when we're in the "inner loop" of development. We want this set for cloud builds (which we can do via environment variable or msbuild parameter in the .yml file), and if you want we could set this here, conditional on release builds.

Reply:
I added condition for "Configuration" == "Release". Not sure if that's enough in your opinion.